### PR TITLE
BAN-79: Add recommended Stencil lint rules

### PR DESCRIPTION
**Pull request by chavlji**
[jira](https://reciprocitylabs.atlassian.net/browse/BAN-79)

Check commit messages for more info.

### CODE REVIEW

File: src/components/zen-animate/zen-animate.tsx [hunk 1]
```diff
@@ -24,12 +25,13 @@ export class ZenAnimate {
   async componentDidRender(): Promise<void> {
     const parent = this.div.querySelector('slot');
     const slot = parent && (parent.assignedNodes()[0] as HTMLElement);
+    if (!slot) return;
```
Comment (f-jauregui) [inline src/components/zen-animate/zen-animate.tsx:28]: Nice!

**Review by f-jauregui — APPROVED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+    project: './tsconfig.json',
   },
   parser: '@typescript-eslint/parser',
   plugins: ['@typescript-eslint'],
   extends: [
     'eslint:recommended',
     'plugin:@typescript-eslint/recommended',
     'prettier/@typescript-eslint',
-    'plugin:prettier/recommended'
+    'plugin:prettier/recommended',
```
Comment (mplesec) [inline .eslintrc.js:11]: Pretty sure this needs to be last.

File: .eslintrc.js [hunk 3]
```diff
@@ -36,5 +37,38 @@ module.exports = {
     'guard-for-in': 0,
     // Allow lexical declarations in case statement:
     'no-case-declarations': 0,
-  }
-}
+
+    // Stencil rules:
```
Comment (mplesec) [inline .eslintrc.js:41]: Aren't these applied by default?

**Review by mplesec — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 3 above)
Comment (chavlji) [reply .eslintrc.js:41]: [Should be](https://github.com/ionic-team/stencil-eslint#recommended-rules) yeah.
But I don't know why it's not working. Eg. if I remove `'@stencil/props-must-be-readonly': 'error',` from specific rules list, it won't show eslint error 🤔

**Pull request merged**